### PR TITLE
chore: simplify process action package

### PR DESCRIPTION
**Pull request by mdelapenya**
- chore: move action package into the process package
- chore: do not expose action struct and methods

## What does this PR do?
It moves everything into the `internal/action` package to the the `internal/process`, converting all its exposed methods, structs and contants into private.

## Why is it important?
Simplify the code, and hide the implementation details of checking a process state, exposing only one public method for that.

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have run the Unit tests (`make unit-test`), and they are passing locally
- [x] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

### CODE REVIEW

**Comment by cachedout** [comment]
@pazone Perhaps it would be good to review this in a screen sharing session with Manu as a part of helping with the transition?

File: internal/process/processes.go [hunk 1]
```diff
@@ -120,7 +120,7 @@ func CheckState(ctx context.Context, deployer deploy.Deployment, service deploy.
 }
 
 // Run executes the command
-func (a *actionWaitProcess) Run(ctx context.Context) (string, error) {
+func (a *actionWait) Run(ctx context.Context) (string, error) {
```
Comment (pazone) [inline internal/process/processes.go:123]: I'm not an expert at go but we could also make `run` function private :)

File: internal/process/processes.go [hunk 2]
```diff
@@ -59,3 +118,288 @@ func CheckState(ctx context.Context, deployer deploy.Deployment, service deploy.
 
 	return nil
 }
+
+// Run executes the command
+func (a *actionWait) Run(ctx context.Context) (string, error) {
```
Comment (pazone) [inline internal/process/processes.go:123]:
```suggestion
func (a *actionWait) run(ctx context.Context) (string, error) {
```

**Review by pazone — APPROVED** [review]

File: internal/process/processes.go (hunk 1 above)
Comment (mdelapenya) [reply internal/process/processes.go:123]: I'm checking the code, and there is an interface in the deploy package, `ServiceOperation`, that exposes a `Run` method. Therefore, this method should be kept public to match the interface.

**Pull request merged**